plugins/module_utils/network/sonic/config/aaa/aaa.py

```diff
@@ -159,52 +155,43 @@ def _state_merged(self, diff):
         :returns: the commands necessary to merge the provided into
                   the current configuration
         """
-        commands = []
-        requests = []
-        if diff:
-            requests = self.get_create_aaa_request(diff)
-            if len(requests) > 0:
-                commands = update_states(diff, "merged")
-        return commands, requests
-
-    def _state_deleted(self, want, have):
-        """ The command generator when state is deleted
+        commands = diff
+        requests = self.get_modify_aaa_requests(commands)
 
-        :rtype: A list
-        :returns: the commands necessary to remove the current configuration
-                  of the provided objects
-        """
-        commands = []
-        requests = []
-        if not want:
-            if have:
-                requests = self.get_delete_all_aaa_request(have)
-                if len(requests) > 0:
-                    commands = update_states(have, "deleted")
+        if commands and len(requests) > 0:
+            commands = update_states(commands, 'merged')
         else:
-            want = utils.remove_empties(want)
-            new_have = self.remove_default_entries(have)
-            d_diff = get_diff(want, new_have, is_skeleton=True)
-            diff_want = get_diff(want, d_diff, is_skeleton=True)
-            if diff_want:
-                requests = self.get_delete_all_aaa_request(diff_want)
-                if len(requests) > 0:
-                    commands = update_states(diff_want, "deleted")
+            commands = []
+
         return commands, requests
 
-    def _state_replaced(self, diff):
+    def _state_replaced(self, want, have, diff):
         """ The command generator when state is merged
 
         :rtype: A list
         :returns: the commands necessary to merge the provided into
                   the current configuration
         """
         commands = []
+        mod_commands = []
         requests = []
-        if diff:
-            requests = self.get_create_aaa_request(diff)
-            if len(requests) > 0:
-                commands = update_states(diff, "replaced")
+        replaced_config = self.get_replaced_config(want, have)
+
+        if replaced_config:
+            is_delete_all = replaced_config == have
```

Comment:
There are many lists in args like have, want and replaced_config, if the lists are not sorted, The comparison "==" does nor work properly???